### ISSUE TITLE
HomeKit (matter) work

### ISSUE THREAD

**Issue by revk** [revk/ESP32-Faikin#93]
Try and get matter working with Faikin

<img width="578" alt="Screenshot 2023-08-15 at 14 40 43" src="https://github.com/revk/ESP32-Faikin/assets/996983/77356820-da8b-4489-8e42-33e90f07be6d">

**Comment by revk** [comment]
I have managed to build a test matter app and load on a test board. This is progress. I have yet to get as far as any integration with Faikin, yet.

**Comment by revk** [comment]
This may be complicated, it looks like some specific flash partitions may be needed, and possibly even some per device "factory set up", which would be a bugger.

**Comment by revk** [comment]
This is a nightmare, it seems that the `esp-matter` IDF itself has a load of things like WS1812 drivers, and when trying to change the example LED driver it links in those drivers regardless rather than letting me use the later managed component. This makes no sense - why would such a driver be ing the matter IDF anyway?

I may have to wait for matter to be more mature, a simple `idf.py add-dependency asp-matter` would be lovely, but we do not seem close to that at the moment.

I'll carry on working on trying to integrate matter in to my build systems anyway.

**Comment by njh** [comment]
It does sound like a maturity problem. Are there any alternatives?
Thanks for looking into it!

**Comment by revk** [comment]
Agreed, I have been looking for a while, so not really sure yet. Still working on it.

**Comment by revk** [comment]
Driving me mad, but getting closer every day.

**Comment by njh** [comment]
I am wondering how long it will be until you decide it would be easier to just write your own Matter library 😀

**Comment by revk** [comment]
I have made a lot of progress testing this.

Key issues remain

- Ensuring that it can work without matter in the same way as now, with WiFI AP set up, and controls for WiFi via the web interface.
- Ensuring MDNS works for simple `name.local` which currently clashes with Matter system.

If I can make it all work I can upgrade current code. For now I am going to be working on a different build for this, which will be available for people to test with soon.

**Comment by njh** [comment]
> If I can make it all work I can upgrade current code. For now I am going to be working on a different build for this, which will be available for people to test with soon.

Good work! I am happy to give it a go, once you have a build ready for testing 👍

**Comment by revk** [comment]
OK coming back to this - I now have a nice reference for someone doing things the way I want to do them (i.e. I control wifi, etc, but allow matter as well). This should help a lot.

**Comment by Cptmeme** [comment]
hi, I just bought one of your boards from amazon and was wondering if there were any updates regarding matter support? (not trying to be annoying, just interested) Thanks for this entire project!!

**Comment by revk** [comment]
Slightly stalled for now, but still planned.

**Comment by Cptmeme** [comment]
Okay great, thank you!

**Comment by revk** [comment]
It is going on hold, sorry. One day, but to be honest, right now, HA seems a better choice for a lot of people.

**Comment by benwa** [comment]
Is there any code for this? Hoping things can pick back up after May of 2024.

**Comment by revk** [comment]
Somewhat stalled, sorry. I managed a basic proof of concept for matter, but we could never sort a certified matter install, and people using HA so much has kind of taken over (and HA can then work with matter).

**Comment by benwa** [comment]
Are you able to share the POC? I'm interested to see if I can get it up and running.

**Comment by revk** [comment]
The PoC was not Faikin code, just using my library on my LED project, sorry. Probably not helpful.

**Comment by benwa** [comment]
Ah, I understand. Thank you though!

**Comment by qstyler** [comment]
Can we maybe boost this work somehow, maybe with a donation or something?

**Comment by revk** [comment]
Not sure that would help - it looks a lot like Matter is a bit of a closed club and a lot of hassle to do properly, and very difficult for an open source project.

**Comment by Cptmeme** [comment]
If people are still interested in matter support for their daikin A/C you can check out a simple esp-matter based project I made to control my daikin a/c, it lacks a lot of features that are in this program so it might not be interesting to most people, but it allows for basic control in all matter ecosystems. (@revk, I know this is a bit of self-promotion, if its inappropriate please accept my apology in advance and feel free to remove the comment, I only wish to help people, and maybe get them off your back about matter support ;) )

**Comment by revk** [comment]
Well done. My focus on matter was really before home assistant integration took off. Give that HA will integrate with Matter, all it takes, AFAIK, is HA to tie it all together, which seems a pretty good compromise really.

**Comment by Cptmeme** [comment]
True, though some people (like me) want to eliminate as many points of failure as possible (i.e. hubs), so for me not having to run homeassistant was not a must but definitely a bonus! My project also allows for thread support which also allowed me to get closer to another goal I have: eliminate all wifi smart home devices and only have them use thread/zigbee/zwave etc.

Edit: here is the link to my repo: https://github.com/Cptmeme/ESP-Matter-Daikin-S21

**Comment by revk** [comment]
Fair enough, and thanks for letting me know. Not completely ruled out, but not a huge priority, sorry.

**Comment by Cptmeme** [comment]
No worries, for now I am happy with the integration I made!
